cli/cppcheckexecutor.cpp

```diff
@@ -499,16 +501,16 @@ void CppCheckExecutor::reportStatus(std::size_t fileindex, std::size_t filecount
 void CppCheckExecutor::reportErr(const ErrorMessage &msg)
 {
     if (mShowAllErrors) {
-        reportOut(msg.toXML());
+        mReport->addFinding(msg);
```

Comment:
I suggest that addFinding returns a string. (SARIF reporter will return an empty string).

Then you can write `reportOut(mReport->addFinding(msg));`